spec/core/ics-033-multi-hop/README.md

```diff
@@ -0,0 +1,543 @@
+        proofHeight := nextChain.GetClientStateHeight(currentChain.ClientID)
+        abortTransactionUnless(connectionHeight >= proofHeight) // ensure that nextChain's client state is update to date
+
+        // the connection end on the current chain representing the previous chain.
+        connectionEnd := currentChain.GetConnection()
+
+        // the connection key
+        connectionKey := host.ConnectionKey(currentChain.ConnectionID())
+
+        // query proof of the currentChain's connection with the previousChain at nextHeight
+        consensusProof, _ := currentChain.QueryProofAtHeight(connectionKey, proofHeight.GetRevisionHeight())
+
+        // create the merkle path with the chain's prefix key
+        prefixedKey := currentChain.GetMerklePath(connectionKey)
+
+        // prepend the proofs so they are ordered from receiving end towards sending end
+        proofs = append(
+            &ProofData{
+                Key:   &prefixedKey,
+                Value: connectionEnd,
+                Proof: connectionProof,
+            },
+            proofs
+        )
+    }
+    return proofs
+}
+```
+
+### Multi-hop Proof Verification Steps
+
+The following outlines the general proof verification steps specific to a multi-hop IBC message.
+
+1. Unpack the multihop proof bytes into consensus states, connection states and channel/commitment proof data.
+2. Check the counterparty client on the receiving end is active and the client height is greater than or equal to the proof height.
+3. Iterate through the connections states to determine the maximum `delayPeriod` for the channel path and verify that the counterparty consensus state on the receiving chain satisfies the delay requirement.
+4. Iterate through connection state proofs and verify each connectionEnd is in the OPEN state and check that the connection ids match the channel connectionHops.
+5. Verify the intermediate state proofs. Starting with known `ConsensusState[N-1]` at the given `proofHeight` on `ChainN` prove the prior chain's consensus and connection state.
+6. Verify that the client id in each consensus state proof key matches the client id in the ConnectionEnd in the previous connection state proof.
+7. Repeat step 3, proving `ConsensusState[i-1]`, and `Conn[i-1,i]` until `ConsensusState[0]` on the source chain is proven.
+   - Start with i = N-1
+   - ConsensusState <- ConsensusProof[i-1].Value on Chain[i]
+   - ConnectionEnd <- ConnectionProof[i-1].Value on Chain[i]
+   - Verify ParseClientID(ConsensusProofs[i].Key) == ConnectionEnd.ClientID
+   - ConsensusProofs[i].Proof.VerifyMembership(ConsensusState.GetRoot(), ConsensusProofs[i].Key, ConsensusProofs[i].Value)
+   - ConnectionProofs[i].Proof.VerifyMembership(ConsensusState.GetRoot(), ConnectionProofs[i].Key, ConnectionProofs[i].Value)
+   - Set ConsensusState from unmarshalled ConsensusProofs[i].Value
+8. Finally, prove the expected channel or packet commitment in `ConsensusState[0]`
```

Comment:
These comments should be updated to reflect that first proof is for the receiving chain and last for the source (or misbehaving) chain.